lib/AST/ASTImporter.cpp

```diff
@@ -282,6 +282,34 @@ namespace clang {
         ToD->setImplicit();
     }
 
+    void addDeclToContexts(Decl *FromD, Decl *ToD) {
+      if (!Importer.isMinimalImport()) {
+        bool Added = false;
+        if (FromD->getDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (ToD->getDeclContext() != ToD->getLexicalDeclContext() &&
+            FromD->getLexicalDeclContext()->containsDeclAndLoad(FromD)) {
+          ToD->getLexicalDeclContext()->addDeclInternal(ToD);
+          Added = true;
+        }
+        if (auto *FromTD = dyn_cast<TagDecl>(FromD)) {
+          auto *ToTD = cast<TagDecl>(ToD);
+          if (!Added &&
```

Comment:
I think there may be cases when `Added` is true, and `FromTD->getLookupParent()->lookup(FromTD->getDeclName()).empty()` is false. (Because the DC's list of contained decl's may be independent from their LookupPtr.)
I think that this may be independent from `Added`, but then the `Added` variable is not needed.
I may be wrong but then we should at least add an assert that no such case exists.